### PR TITLE
Create CSV Report Generator.

### PR DESCRIPTION
**Pull request by tpendragon**
Work towards #5631, might be useful for a more generic report generator.

### CODE REVIEW

File: app/values/csv_report.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    attr_reader :resource
+    def initialize(resource)
+      @resource = resource
+    end
+
+    def to_h
+      Hash[
+        # __attributes__ is much faster and doesn't include nil values.
+        resource.__attributes__.merge(attributes_without_reserved(imported_metadata)).map do |key, values|
+          values = Array.wrap(values).map(&:to_s).join(", ")
```
Comment (hackartisan) [inline app/values/csv_report.rb:45]: Is this separating values with a comma? Can we do that with a CSV? Maybe add a field to the test with multiple values.

**Review by hackartisan — CHANGES_REQUESTED** [review]
Yes, this looks really handy! I'd be interested in seeing how a refactor of https://github.com/pulibrary/figgy/blob/fcd0e83eaa77e27c7a78038a4a94ea35de0761a6/app/controllers/reports_controller.rb would look. I'm undecided on blocking for that because I worry it would get lost if we don't do it now and I don't want equivalent code everywhere, but I recognize that this ticket is asap. What do you think @tpendragon?

**Comment by hackartisan** [comment]
And this is really nicely commented thank you for that

**Comment by tpendragon** [comment]
> I'm undecided on blocking for that because I worry it would get lost if we don't do it now and I don't want equivalent code everywhere, but I recognize that this ticket is asap. What do you think @tpendragon?

I looked at that too. I didn't tackle it because the reality is I'm out of time - I don't think I'll be able to do that this week.

**Comment by hackartisan** [comment]
Yeah, that makes sense. It might be an interesting refactor for a student to work on and we have room on that board for sure.

**Comment by tpendragon** [comment]
@hackartisan I added the extent field, which has a comma in it. Work?

**Review by hackartisan — APPROVED** [review]

**Comment by hackartisan** [comment]
i created https://github.com/pulibrary/figgy/issues/5687

**Pull request merged**